### PR TITLE
bpo-27165: Skip callables when printing an exception details

### PR DESCRIPTION
**Pull request by BookLaugh**
Included 2 tests and a news entry for the change. 

Since I deleted my previous forked cpython repo, I can't add commits to PR #9695 - so I'm creating new PR with the same change to code, but tests and news entry added.

https://bugs.python.org/issue27165

### CODE REVIEW

File: Lib/test/test_cgitb.py [hunk 1]
```diff
@@ -63,6 +63,25 @@ def test_syshook_no_logdir_text_format(self):
         self.assertNotIn('<p>', out)
         self.assertNotIn('</p>', out)
 
+    def test_exception_text(self):
+        # Issue 27165: Exclude callables from exception details
+        try:
+            raise Exception("text foo")
+        except Exception as e:
+            exc_text = cgitb.text(sys.exc_info())
+            print(exc_text)
```
Comment (tirkarthi) [inline Lib/test/test_cgitb.py:72]: Print statement is not needed here and in below. I added it for showing the difference in output between the PR changes.

Comment (tirkarthi) [reply Lib/test/test_cgitb.py:72]: You can add another assertion for `__doc__` just to make sure non-callables are present. I will wait for the reviewers to see if any more cases need to be added. Thanks.

**Review by serhiy-storchaka — COMMENTED** [review]
The html variant skips names starting with an underscore. Shouldn't they be skipped in the raw text variant?

Test also methods not starting with an underscore (currently there is only one such method: `with_traceback`) and instance attributes starting with an underscore like `__dict__`.

Add tests for exceptions that have attributes like `UnicodeEncodeError` or `ImportError`.

**Comment by iritkatriel** [comment]
cgi/cgitb are deprecated as per [PEP 594](https://www.python.org/dev/peps/pep-0594/), so there won't be further enhancements to them.

**Pull request closed without merging**